provider/oracle/shape.go

```diff
@@ -0,0 +1,100 @@
+package oracle
+
+import (
+	"github.com/hoenirvili/go-oracle-cloud/response"
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/constraints"
+)
+
+type shape struct {
+	// name of the shape
+	name string
+	// cpu field shows the number of CPU threads
+	cpus uint64
+	// ram field shows the amount of memory in MB
+	ram uint64
+}
+
+// find shape searches through the oracle api and finds the best shape
+// that is complaint with the bootstrap constraints containing the ram and cpu
+func findShape(shapes []response.Shape, cons constraints.Value) (*shape, error) {
```

Comment:
We already have some logic around selecting an image and instance type; please use that. See environs/instances.FindInstanceSpec. Convert the "shapes" into instances.InstanceType structures. See provider/ec2/image.go for an example.